packages/plugin-autocapture-browser/src/data-extractor.ts

```diff
@@ -194,7 +194,11 @@ export class DataExtractor {
 
   combineText = (element: Element): string => {
     let text = '';
-    if (isNonSensitiveElement(element) && element.childNodes && element.childNodes.length) {
+    // Check if element or any parent has data-amp-mask attribute
+    const hasMaskAttribute = element.closest('[' + constants.DATA_AMP_MASK_ATTRIBUTE + ']') !== null;
```

Comment:
This causes parents to be rechecked repeatedly, should revisit a different solution.

```
Check parent have data-amp-mask
If yes, return null.

el.querySelector(‘[data-amp-mask]’)

If found, do more complex logic.
Else return innerText.
```

if a child has data-amp-mask:
cloneNode on target element 
clonedTree.querySelectorAll(‘[data-amp-mask]’)
remove all found
cnoedTree.innerText
```


```

Reply:
Updated